wallet-extension/src/background/storage/state.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  createClient,
+  createFaucetClient,
+  createSigner,
+  initializeOrDeployWallet,
+} from "../../features/blockchain";
+import { initializeActivities } from "../../features/store/model/activities.ts";
+import {
+  setFaucetClient,
+  setPublicClient,
+  setSigner,
+} from "../../features/store/model/blockchain.ts";
+import { setEndpoint } from "../../features/store/model/endpoint.ts";
+import { setPrivateKey } from "../../features/store/model/privateKey.ts";
+import { setExistingWallet, setIsWalletInitialized } from "../../features/store/model/wallet.ts";
+
+// Saves other blockchain fields to Chrome storage
+export async function saveUserDetails(fields: {
+  rpcEndpoint: string;
+  shardId: number;
+  privateKey: string;
+  walletAddress: string;
+}): Promise<void> {
+  try {
+    await chrome.storage.local.set({ blockchainFields: fields });
+    console.log("Blockchain fields saved:", fields);
+  } catch (error) {
+    console.error("Error saving blockchain fields:", error);
+
+    // Rethrow the error to propagate it to the caller
+    throw new Error(`Failed to save blockchain fields to Chrome storage`);
+  }
+}
+
+// Initializes blockchain resources by loading fields from Chrome storage and setting up clients
+export async function initializeFromStorageAndSetup(): Promise<void> {
+  try {
+    const { blockchainFields } = await chrome.storage.local.get("blockchainFields");
+    if (blockchainFields) {
+      console.log("Loaded blockchain fields from storage: ", blockchainFields);
+
+      const { rpcEndpoint, shardId, privateKey, walletAddress } = blockchainFields;
```

Comment:
strange that you don't need to `JSON.parse()` cz local storage stores everything as strings.

Reply:
`chrome.storage.local.get()` automatically handles JSON serialization and deserialization